doitlive.py

```diff
@@ -181,6 +184,18 @@ class TTY(object):
     DIM = ANSICode(dim=True)
 
 
+def echo(message=None, file=None, nl=True, err=False, color=None, carriage_return=False):
+    """
+    Patched click echo function.
+    """
+    if carriage_return and nl:
+        click_echo(message + '\r\n', file, False, err, color)
```

Comment:
This will not be compatible with older versions (< 4.0) of click, though I am fine with bumping up the click version requirement. Can you change that in this PR?